services/blockchain-indexer/shared/dataService/business/events.js

```diff
@@ -70,6 +71,31 @@ const getEventsByHeight = async (height) => {
 	return eventsFromNode;
 };
 
+const getEventsByBlockID = async (blockID) => {
+	// Get from cache
+	const cachedEvents = await eventCacheByBlockID.get(blockID);
+	if (cachedEvents) return JSON.parse(cachedEvents);
+
+	// Get from DB only when isPersistEvents is enabled
+	if (config.isPersistEvents) {
```

Comment:
The `if` condition can be removed. We also maintain the events for non-finalized blocks even if `config.isPersistEvents` is not set.